boa/src/builtins/iterable/mod.rs

```diff
@@ -0,0 +1,141 @@
+use crate::builtins::function::{BuiltInFunction, Function, FunctionFlags};
+use crate::builtins::ArrayIterator;
+use crate::object::{Object, PROTOTYPE};
+use crate::BoaProfiler;
+use crate::{property::Property, Context, Value};
+
+#[derive(Debug, Default)]
+pub struct IteratorPrototypes {
+    iterator_prototype: Value,
+    array_iterator: Value,
```

Comment:
```suggestion
    iterator_prototype: GcObject,
    array_iterator: GcObject,
```
It would be better if these were `GcObject` since:
 - The prototype can only be an object (fits the GcObject) while `Value` can be other variants
 - `Value` is 24 bytes a but `GcObject` is 8